src/datatype/index.js

```diff
@@ -172,26 +177,73 @@ export class DataType extends TypedEmitter {
 
   /**
    * @param {string} docId
+   * @param {{ lang?: string }} [opts]
    */
-  async getByDocId(docId) {
+  async getByDocId(docId, { lang } = {}) {
     await this.#dataStore.indexer.idle()
-    const result = this.#sql.getByDocId.get({ docId })
+    const result = /** @type {MapeoDoc} */ (this.#sql.getByDocId.get({ docId }))
     if (!result) throw new NotFoundError()
-    return deNullify(result)
+    return this.#translate(deNullify(result), { lang })
   }
 
-  /** @param {string} versionId */
-  async getByVersionId(versionId) {
-    return this.#dataStore.read(versionId)
+  /**
+   * @param {string} versionId
+   * @param {{ lang?: string }} [opts]
+   */
+  async getByVersionId(versionId, { lang } = {}) {
+    const result = await this.#dataStore.read(versionId)
+    return this.#translate(result, { lang })
+  }
+
+  /**
+   * @param {MapeoDoc} doc
+   * @param {{ lang?: string }} [opts]
+   */
+  async #translate(doc, { lang } = {}) {
+    if (!lang) return doc
+
+    const { language, region } = parseBCP47(lang)
+    if (!language) return doc
+    // const translatedDoc = JSON.parse(JSON.stringify(doc))
+
+    let value = {
+      languageCode: language,
+      schemaNameRef: doc.schemaName,
+      docIdRef: doc.docId,
+      regionCode: region !== null ? region : undefined,
+    }
+    let translations = await this.#translation.get(value)
+    // if passing a region code returns no matches,
+    // fallback to matching only languageCode
+    if (translations.length === 0) {
+      value.regionCode = undefined
+    }
+    translations = await this.#translation.get(value)
```

Comment:
This does an extra lookup. To avoid that:

```suggestion
    if (translations.length === 0 && value.regionCode) {
      value.regionCode = undefined
      translations = await this.#translation.get(value)
    }
```